lib/Doctrine/ORM/Tools/Pagination/Paginator.php

```diff
@@ -156,7 +155,7 @@ public function getIterator()
             $ids          = array_map('current', $foundIdRows);
 
             $this->appendTreeWalker($whereInQuery, WhereInWalker::class);
-            $whereInQuery->setHint(WhereInWalker::HINT_PAGINATOR_ID_COUNT, count($ids));
+            $whereInQuery->setHint(WhereInWalker::HINT_PAGINATOR_HAS_IDS, true);
```

Comment:
Could I please get an extra pair of 👀 on this? 

Once we reach that point, `$ids` is never empty, right?